src/JSInterop/Microsoft.JSInterop/src/Infrastructure/DotNetDispatcher.cs

```diff
@@ -350,15 +350,8 @@ public static void ReceiveByteArray(JSRuntime jsRuntime, int id, byte[] data)
 
     private static (MethodInfo, Type[]) GetCachedMethodInfo(AssemblyKey assemblyKey, string methodIdentifier)
     {
-        if (string.IsNullOrWhiteSpace(assemblyKey.AssemblyName))
-        {
-            throw new ArgumentException($"Property '{nameof(AssemblyKey.AssemblyName)}' cannot be null, empty, or whitespace.", nameof(assemblyKey));
```

Comment:
Nit: `Property '{nameof(AssemblyKey.AssemblyName)}'` is slightly clearer than just `nameof(AssemblyKey.AssemblyName)` because it makes clearer that the property and not the paremeter itself is null. I know `AssemblyKey` cannot be null since it's a struct, but people running into this exception might not. `'{nameof(AssemblyKey)}.{nameof(AssemblyKey.AssemblyName)}'` would be even better.

I think everything else between `CascadingValue<TValue>.SetParametersAsync` and here in the PR looks fine even if it changes the wording slightly because it's more obvious what's invalid.

Reply:
I understand the point but since the entire expression will be captured as parameter name, I think it's clear enough that it's a property:

`System.ArgumentNullException: Value cannot be null. (Parameter 'assemblyKey.AssemblyName')`

Do you still want me to revert this one?

Reply:
No need to revert. Thanks for pushing back since the new message is clearer than I assumed. And thanks for doing all this work!